extensions/integration-tests/src/notebook.test.ts

```diff
@@ -18,122 +18,159 @@ import * as fs from 'fs';
 if (context.RunTest) {
 	suite('Notebook integration test suite', function () {
 		setup(function () {
-			console.log(`Start "${this.currentTest.title}"`);
+			this.NbTester = new NotebooksTester(this.currentTest.title);
```

Comment:
Whats the goal of moving this out into a class? seems unnecessary

Reply:
decorators can only be applied to classes and class members so any test that needs to be stressified (or perfified in future) needs to be have core logic in a class.

Reply:
Could you just use a https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators instead on the function that is passed to run the test?